src/test/java/edu/ucsb/cs156/rec/controllers/RecommendationRequestControllerTest.java

```diff
@@ -0,0 +1,339 @@
+
+        RecommendationRequest rec2 = RecommendationRequest.builder()
+                .professorName("Singh")
+                .professorEmail("ambuj@cs.ucsb.edu")
+                .requesterName("Sara")
+                .recommendationTypes("rec type")
+                .details("details")
+                .status("status")
+                .submissionDate(LocalDateTime.parse("2022-01-03T00:00:00"))
+                .completionDate(LocalDateTime.parse("2022-01-03T00:00:00"))
+                .build();
+
+        ArrayList<RecommendationRequest> expectedRecommendationRequests = new ArrayList<>();
+        expectedRecommendationRequests.addAll(Arrays.asList(rec1, rec2));
+
+        when(recommendationRequestRepository.findAll()).thenReturn(expectedRecommendationRequests);
+
+        // act
+        MvcResult response = mockMvc.perform(get("/api/recommendationrequests/all"))
+                .andExpect(status().isOk()).andReturn();
+
+        // assert
+        verify(recommendationRequestRepository, times(1)).findAll();
+        String expectedJson = mapper.writeValueAsString(expectedRecommendationRequests);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedJson, responseString);
+    }
+
+    @WithMockUser(roles = { "ADMIN", "USER" })
+    @Test
+    public void an_admin_user_can_post_a_new_recommendation_request() throws Exception {
+        // arrange
+        RecommendationRequest rec1 = RecommendationRequest.builder()
+                .professorName("Conrad")
+                .professorEmail("phtcon@ucsb.edu")
+                .requesterName("Amanda")
+                .recommendationTypes("rec type")
+                .details("details")
+                .status("status")
+                .submissionDate(LocalDateTime.parse("2022-01-03T00:00:00"))
+                .completionDate(LocalDateTime.parse("2022-01-03T00:00:00"))
+                .build();
+
+        when(recommendationRequestRepository.save(eq(rec1))).thenReturn(rec1);
+
+        // act
+        MvcResult response = mockMvc.perform(
+                post("/api/recommendationrequests/post?professorName=Conrad&professorEmail=phtcon@ucsb.edu&requesterName=Amanda&recommendationTypes=rec type&details=details&status=status&submissionDate=2022-01-03T00:00:00&completionDate=2022-01-03T00:00:00")
+                        .with(csrf()))
+                .andExpect(status().isOk()).andReturn();
+
+        // assert
+        verify(recommendationRequestRepository, times(1)).save(rec1);
+        String expectedJson = mapper.writeValueAsString(rec1);
+        String responseString = response.getResponse().getContentAsString();
+        assertEquals(expectedJson, responseString);
+    }
+
+    @WithMockUser(roles = { "ADMIN", "USER" })
+    @Test
+    public void admin_can_delete_a_recommendation_request() throws Exception {
+        // arrange
+        RecommendationRequest recReq = RecommendationRequest.builder()
```

Comment:
I am not entirely sure, but shouldn't this include .id(15L) like you did in the tests below? I think that would make sense since we are searching the recommendation request repository for an id with 15 to delete.